system-addon/content-src/lib/init-store.js

```diff
@@ -44,12 +44,27 @@ const messageMiddleware = store => next => action => {
 };
 
 const rehydrationMiddleware = store => next => action => {
+  // XXX Due to the internal mechanism of redux middleware, we had to add this property in here, see
+  // more details in #3725. We should refactor this workaround via fixing #3725.
+  if (!store._earlyMessageQueue) {
```

Comment:
I would put this after the `store._didRehydrate` check, since it will save a check for most actions